apps/rush-lib/src/schemas/build-cache.schema.json

```diff
@@ -3,17 +3,10 @@
   "title": "Configuration for Rush's build cache.",
```

Comment:
This configuration is relatively simple.  I think we should try to avoid a proliferation of too many small files, particular for project-specific settings (where we would need to parse the file separately for every project, even if it comes from a rig).

Thinking about the rubric, these are justifiable reasons to make separate files:
- A file that can collect an arbitrarily large amount of user  data (e.g. **browser-approved-packages.json**).  The `"projects"` section of **rush.json** really should really be split into a separate file under this rationale.
- A file that represents a complex self-contained world (e.g. **command-line.json**)
- A file that accumulates merge conflicts and they can all be resolved together (e.g. **repo-state.json**).
- A unit that can be duplicated with different names (e.g. **scenario-template.json** for `rush deploy`)

Whereas Rush features don't seem to make very good boundaries.  For example, **version-policies.json** and **common-versions.json* maybe should be rolled together, maybe right into **rush.json**.

Reply:
🤔

@iclanton  What if we rename `build-cache.json` to be:

 `<your-project>/config/rush-project.json`

This file would hold all of Rush's project-specific settings.  (If one of the above criterion arises in the future, the naming pattern would be like ``<your-project>/config/rush-deploy.json`.)  This approach gives us a unique name `rush-project` that the docs can refer to when talking about this file.